server/src/main/java/org/elasticsearch/plugins/internal/DocumentParsingProvider.java

```diff
@@ -13,35 +13,27 @@
  */
 public interface DocumentParsingProvider {
     DocumentParsingProvider EMPTY_INSTANCE = new DocumentParsingProvider() {
-        @Override
-        public DocumentSizeObserver newDocumentSizeObserver() {
-            return DocumentSizeObserver.EMPTY_INSTANCE;
-        }
-
-        @Override
-        public DocumentSizeReporter getDocumentParsingReporter(String indexName) {
-            return DocumentSizeReporter.EMPTY_INSTANCE;
-        }
-
-        @Override
-        public DocumentSizeObserver newFixedSizeDocumentObserver(long normalisedBytesParsed) {
-            return DocumentSizeObserver.EMPTY_INSTANCE;
-        }
     };
 
     /**
      * @return a new 'empty' observer to use when observing parsing
      */
-    DocumentSizeObserver newDocumentSizeObserver();
+    default DocumentSizeObserver newDocumentSizeObserver() {
+        return DocumentSizeObserver.EMPTY_INSTANCE;
+    }
 
     /**
      * @return an observer with a previously observed value (fixed to this value, not continuing)
      */
-    DocumentSizeObserver newFixedSizeDocumentObserver(long normalisedBytesParsed);
+    default DocumentSizeObserver newFixedSizeDocumentObserver(long normalisedBytesParsed) {
+        return DocumentSizeObserver.EMPTY_INSTANCE;
+    }
 
     /**
      * @return an instance of a reporter to use when parsing has been completed and indexing successful
      */
-    DocumentSizeReporter getDocumentParsingReporter(String indexName);
+    default DocumentSizeReporter getDocumentParsingReporter(String indexName) {
```

Comment:
should this be `newDocumentSizeReporter` to be consistent with the others?

Reply:
Wondering, looking at the final signature of `DocumentSizeReporter`, it seems much more general than a doc size reporter. Is the name still correct?

Reply:
> should this be newDocumentSizeReporter to be consistent with the others?
++ will change

> Wondering, looking at the final signature of DocumentSizeReporter, it seems much more general than a doc size reporter. Is the name still correct?
you mean the addition of `onParsingCompleted` ? I think that still is about reporting a doc size right?

Reply:
>  I think that still is about reporting a doc size right?

That's how it's eventually used, but the signature is totally independent of that.
`onIndexingCompleted(ParsedDocument parsedDocument)` is more more general than doc size.
But same above, struggling myself to come up with better names, we can revisit if we think it's necessary.